### DISCUSSION TITLE
Removing tag pairs from Antlers tags so logic still works with runtime parser

### DISCUSSION THREAD

**Discussion by bruceontheloose** [statamic/cms#9113] [Troubleshooting] [answered]
Hello,

I'm in the process of trying to switch a project from the legacy parser to the runtime parser. I've run into an error (in several files) that's preventing hundreds of pages from rendering:

  > Unclosed "if" control structure. Tag pairs are not supported within Antlers tags.

Here's an example that works with the legacy parser but gives that error with the runtime parser:

```antlr
{{ if total_results === 1 }}
  {{ partial:globals/images/hero_image 
      {{ if landing_page_row_item_photo }}
          :image="landing_page_row_item_photo" 
      {{ elseif landing_page_row_item_entry:hero_image }}
          :image="landing_page_row_item_entry:hero_image"
      ...another elseif...
      {{ /if }}
      custom_classes="m-0" }}
{{ else }}
  ...similar, but with more params that all cases share...
{{ /if }}
```
The `if` condition in the `partial` tag is the problem. To fix it, I know I could switch that to the following—

```antlr
{{ if total_results === 1 }}
  {{ if landing_page_row_item_photo }}
    {{ partial:globals/images/hero_image 
        :image="landing_page_row_item_photo" 
        custom_classes="m-0" }}
  {{ elseif landing_page_row_item_entry:hero_image }}
    {{ partial:globals/images/hero_image 
        :image="landing_page_row_item_entry:hero_image" 
        custom_classes="m-0" }}
  ...another elseif...
  {{ /if }}
{{ else }}
  ...similar, but with more params that all cases share...
{{ /if }}
```

—but I'd like to avoid repeating code if possible. (I chose a shorter example to show here. Other instances of ours would have a lot more repetition.)

I suspect there's a better way I'm not aware of. Do you have any suggestions? (I tried a `switch()` on another `partial`; that worked but resulted in less legible code. Similarly, I want to avoid nesting a ternary inside another ternary.)

Thanks very much in advance for any pointers.

Bruce

**Comment by jasonvarga** [comment, marked as answer]
You could set variables and pass them in.

```antlers
{{ image = landing_page_row_item_photo ?? landing_page_row_item_entry:hero_image ?? "fallback" }}

{{ partial:globals/images/hero_image :image="image" custom_classes="m-0" }}
```

**Reply by bruceontheloose** [reply]
Excellent, thank you, @jasonvarga!
